src/lib.rs

```diff
@@ -1,19 +1,137 @@
-//! [Balena](https://www.balena.io) TEMplating ENgine
+//! [balena] **tem**plating **en**gine
 //!
 //! A crate that provides facilities to:
 //!
-//! * evaluate expressions
-//! * evaluate logical expressions
-//! * retrieve AST of any expression for further validation
+//! * parse an expression
+//! * evaluate an expression
+//! * evaluate an expression as a boolean (logical expressions)
+//! * register custom functions and filters
+//!
+//! # Versioning
+//!
+//! This crate is being actively developed and it does NOT follow [Semantic Versioning] yet.
+//! It will follow semantic versioning when it reaches version 1.0.
+//!
+//! MINOR version changes denotes incompatible API changes and PATCH version changes denotes
+//! both new functionality in a backwards-compatible manner and backwards-compatible bug fixes.
+//!
+//! # Examples
+//!
+//! ## Expression parsing
```

Comment:
It takes a bit of time to go through this example - 
it may make sense to reshuffle the example list, for this one to be somewhere further down the line, to use smaller examples first? OR
we could maybe make the expression smaller ? 
The part needing the most brain power for me was that there was `rhs` and `manual` both, maybe renaming `manual` to something like `full_expression` or something like that would help as well ?

Reply:
I just split Examples in two sections - _High level_ and _Low level_ with more notes.